### DISCUSSION TITLE
Error en los tests de venta - Macowins

### DISCUSSION THREAD

**Discussion by LeoCubelli** [dds-jv/foro#208] [Consultas de herramientas] [answered]
Buenas, pude correr los 5 tests que venían en el ejercicio pero al agregar más para testear las ventas y ganancias me tira error en la línea 67
![test](https://user-images.githubusercontent.com/62668099/116337649-d421ce00-a7b0-11eb-9c08-148e0cddfb88.png)

**Comment by nanderson87** [comment, marked as answer]
Hola Leo,

Al parecer aunque el proyecto esta compilando, al correr ese test no puede ejecutar la instrucción de la linea 67 ya que termina en un `null pointer exception`. En lineas generales esto sucede cuando intentas utilizar algun objeto que no esta siendo referenciando es decir que su valor es null. 

Ejemplo:
String unString = null;
unString.length(); _// Esto terminaría en tiempo de ejecución en un Null Pointer Exception._

Tendrías que ver que esta pasando dentro de esa instrucción que este haciendo que en algún momento quieras hacer algo con un objeto que  tiene una referencia a nulll.

1. Podes correr ese test desde el IDE con el botón play que se encuentra al lado del test. Deberías llegar al misma excepción pero te debería mostrar el stacktrace donde podrías ver en más detalle donde esta sucediendo ese null pointer.
2. Si en un principio no detectas la falla. Siempre podes hacer un debug de ese test para ver en más detalle instrucción por instrucción que se va ejecutando y los valores de las variables que entran en juego en cada una de las mismas.

**Reply by LeoCubelli** [reply]
Ahí estuve debugeando y al hacer new Venta() no me inicializa la lista de items y aparece como null. Ahí le agrege el new ArrayList y funco. Gracias Nico
